scripts/generate_presign_url/main.py

```diff
@@ -0,0 +1,138 @@
+import json
+import os
+import sys
+import pandas as pd
+from enum import Enum
+from typing import Dict, List
+from boto3.dynamodb.conditions import Attr
+
+DIR_PATH = os.path.dirname(os.path.realpath(__file__))
+SOURCE_PATH = os.path.join(
+    DIR_PATH, "..", "..", "lambdas", "layers", "util"
+)
+sys.path.append(SOURCE_PATH)
+
+import util
+from util import dynamodb, agha
+
+DYNAMODB_ARCHIVE_RESULT_TABLE_NAME = 'agha-gdr-result-bucket-archive'
+DYNAMODB_ARCHIVE_STAGING_TABLE_NAME = 'agha-gdr-staging-bucket-archive'
+DYNAMODB_ARCHIVE_STORE_TABLE_NAME = 'agha-gdr-store-bucket-archive'
+DYNAMODB_ETAG_TABLE_NAME = 'agha-gdr-e-tag'
+DYNAMODB_RESULT_TABLE_NAME = 'agha-gdr-result-bucket'
+DYNAMODB_STAGING_TABLE_NAME = 'agha-gdr-staging-bucket'
+DYNAMODB_STORE_TABLE_NAME = 'agha-gdr-store-bucket'
+STAGING_BUCKET = 'agha-gdr-staging-2.0'
+RESULT_BUCKET = 'agha-gdr-results-2.0'
+STORE_BUCKET = 'agha-gdr-store-2.0'
+
+CLIENT_S3 = util.get_client('s3')
+
+##################################################################################################################
+# TODO: Update the following information before running the script.
+# TODO: Make sure AWS_PROFILE is set for the script to run
+# NOTE: Only works in store bucket
+"""
+To run the script:
+cd scripts/generate_presign_url
+python3 main.py
+"""
+
+
+def get_argument():
```

Comment:
This means the file will need to be updated for each use case.
Could instead pass arguments to the script?
May have the advantage that some checks are built in... 

```
import argparse

if __name__ == "__main__":

    ################################################################################
    # argument parsing

    parser = argparse.ArgumentParser(description='Generate pre-signed URLs for AGHA files.')
    parser.add_argument('--dryrun',
                        default=False,
                        action='store_true',
                        help="Perform a dry run.")
    parser.add_argument('-t', '--filetype',
                        default=['FASTQ', 'BAM', 'VCF'],
                        nargs='+',
                        choices=['FASTQ', 'BAM', 'VCF'],
                        help='Filetype to filter by (FASTQ, BAM, VCF). Space separated if more than one. Default: all files')
    parser.add_argument('-s', '--study-ids',
                        required=True,
                        nargs='+',
                        help="AGHA study ID(s) to retrieve files for. Space separated if more than one.")
    parser.add_argument('-f', '--flagship',
                        required=True,
                        help="Code of the flagship the sample belongs to.")

    args = parser.parse_args()
    filetype = args.filetype
    study_ids = args.study_ids

    print(f"Filetype: {args.filetype}")
    print(f"Flagship: {args.flagship}")
    print(f"IDs: {args.study_ids}")
    print(f"DryRun? {args.dryrun}")
    print('\n')
    print(args)
    ```

Reply:
Thanks for the drafting it here, I made this into the code.

Initially, I thought would be easier to edit parameters in the code instead of terminal. But agree we should take advantage of built-in `argparse` library.